examples/cpp_api/nullable_attribute.cc

```diff
@@ -0,0 +1,163 @@
+  schema.set_domain(domain).set_order({{TILEDB_ROW_MAJOR, TILEDB_ROW_MAJOR}});
+
+  // Create two attributes "a1" and "a2", the first fixed and the second
+  // variable-sized.
+  Attribute a1 = Attribute::create<int>(ctx, "a1");
+  Attribute a2 = Attribute::create<std::vector<int>>(ctx, "a2");
+
+  // Set both attributes as nullable
+  a1.set_nullable(true);
+  a2.set_nullable(true);
+
+  schema.add_attribute(a1);
+  schema.add_attribute(a2);
+
+  // Create the (empty) array on disk.
+  Array::create(array_name, schema);
+}
+
+template <typename T>
+std::ostream& operator<<(std::ostream& os, std::vector<T> vec) {
+  os << "{";
+  for (unsigned long i = 0; i < vec.size() - 1; i++) {
+    os << +vec[i] << " ";
+  }
+
+  os << +vec.back() << "}";
+  return os;
+}
+
+void write_array() {
+  Context ctx;
+
+  // Prepare some data for the array
+  std::vector<int> a1_data = {100, 200, 300, 400};
+  std::vector<int> a2_data = {10, 10, 20, 30, 30, 30, 40, 40};
+  std::vector<uint64_t> a2_el_off = {0, 2, 3, 6};
+  std::vector<uint64_t> a2_off;
+  for (auto e : a2_el_off)
+    a2_off.push_back(e * sizeof(int));
+
+  // Open the array for writing and create the query
+  Array array(ctx, array_name, TILEDB_WRITE);
+  Query query(ctx, array);
+  query.set_layout(TILEDB_ROW_MAJOR);
+
+  // Specify the validity buffer for each attribute
+  std::vector<uint8_t> a1_validity_buf = {1, 0, 0, 1};
+  std::vector<uint8_t> a2_validity_buf = {0, 1, 1, 1, 0, 0, 1, 1};
+
+  // Print out the data we are writing.
+  std::cout << "Write attribute a1:\n Data: " << a1_data
+            << "\n Validity: " << a1_validity_buf << "\n";
+  std::cout << "Write attribute a2:\n Data: " << a2_data
+            << "\n Validity: " << a2_validity_buf << "\n Offsets: " << a2_off
+            << "\n";
+
+  // Set the query buffers specifying the validity for each data
+  query.set_buffer_nullable("a1", a1_data, a1_validity_buf)
+      .set_buffer_nullable("a2", a2_off, a2_data, a2_validity_buf);
+
+  // Perform the write and close the array.
+  query.submit();
+  array.close();
+}
+
+void read_array() {
+  Context ctx;
+
+  // Prepare the array for reading
+  Array array(ctx, array_name, TILEDB_READ);
+
+  // Prepare the vectors that will hold the results
+  std::vector<int> a1_data(4);
+  std::vector<uint8_t> a1_validity_buf(a1_data.size());
+
+  std::vector<int> a2_data(8);
+  std::vector<uint64_t> a2_off(4);
+  std::vector<uint8_t> a2_validity_buf(a2_data.size());
+
+  // Prepare and submit the query, and close the array
+  Query query(ctx, array);
+  query.set_layout(TILEDB_ROW_MAJOR);
+
+  // Read the full array
+  const std::vector<int> subarray_full = {1, 2, 1, 2};
+  query.set_subarray(subarray_full);
+
+  // Set the query buffers specifying the validity for each data
+  query.set_buffer_nullable("a1", a1_data, a1_validity_buf)
+      .set_buffer_nullable("a2", a2_off, a2_data, a2_validity_buf);
+  query.submit();
+  array.close();
+
+  // Print out the results.
```

Comment:
I'm thinking the right thing to do here would be to print the values, substituting null values as the "NULL" string. So we could do something like:
```
std::cout << "a1: ";
for (i = 0; i < 4; ++i)
  std::cout << (a1_validity_buf[i] > 0 ? std::to_string(a1_data[i]) : "NULL");
```

Which should print "a1: 100 NULL NULL 400".

Reply:
Nice, thanks